### ISSUE TITLE
Native Spot Termination Handling: Make Spot Rebalance Recommendations Optional

### ISSUE THREAD

**Issue by snorlaX-sleeps** [aws/karpenter#2853]
### Tell us about your request

When Native Spot Termination Handling (NSTH from here on) supports Spot Rebalance recommendations in the future, please make it an optional flag for the deployment. (Just going by the docs it sounds like it may eventually be implemented)

### Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?

While Spot Rebalance Recommendations can be useful to act upon for many clusters, for some they are highly detrimental.

We run several clusters where the majority of our pods are statefulsets, which require mounting EBS volumes that are tied to a specific AZ.
What happens is:
- A node will be terminated according to a rebalance event
-  Karpenter will bring up a node to provide the EBS volume to the pod
- AWS will send another rebalance recommendation

This leads to a circle / cycle of node terminations.

### Are you currently working around this issue?

Not deploying the `aws_cloudwatch_event_rule` for rebalance recommendations to a cluster this affects, while still supporting the others (scheduled change, spot termination, instance state).
Basically removing those events from the queue.

### Additional Context

We are still running `v0.18.x` and have yet to test this out.
Looking forward to rolling out the native implementation instead of using a separate service.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by bwagner5** [comment]
Spot Rebalance Recommendations will absolutely be optional in Karpenter when they are implemented since they are not recommended for the majority of Spot use-cases.

**Comment by snorlaX-sleeps** [comment]
Thanks @bwagner5 - I didn't want to pollute #2813 with my request as it was slightly different :)
Happy for it to be closed as it's going to be optional :)
